### DISCUSSION TITLE
Converting Word doc/tables to tabularray-flavored LaTeX

### DISCUSSION THREAD

**Discussion by russurquhart1** [lvjr/tabularray#524] [Q&A]
Hi All,
My predecessors had created some really long Word documents with equally long tables, also encoded in Word. (My boss and I were in agreement, that this should not have been done, as IMO, Word can be shaky, etc. But we work with what we have.)
I'd like to use some tool(s) to have the tables convert to LaTeX using the tabularray package. Most of the translators I've seen make each cell into a minipage object. I can see the reasoning, but it just seems kind of clunky to me.
So I'm asking the experts here how they would convert such tables to LaTeX and the tabularray package, short of by hand manual recreation?

Thanks for any suggestions/help!

Russ

**Comment by muzimuzhi** [comment]
I don't have any experience in converting WORD tables into LaTeX ones.

Have you tried [`pandoc`](https://github.com/jgm/pandoc)?

> Most of the translators I've seen make each cell into a minipage object.

Or you can filter (with the help of regular expression, perhaps) the conversion results to remove the minipage environments.

**Reply by russurquhart1** [reply]
Pandoc, does a similar kind of conversion using minipages. 

I think I found something. There is an XSLT that you can apply as a transform, when you save the Word docs as a 2003 .xml file. The transform, convert the Word doc, as best as it can, to .tex. It, however, uses a similar minipage structure. However, as the XSLT is a text file, and I've had some experience with XSLT's, I can see where it is writing the minipage code, and modify that to something that tabularray is more comfortable with. I just found this, so I'll be trying it and let everyone know what I find! Thanks!
